src/cli/task.rs

```diff
@@ -309,20 +324,48 @@ pub fn execute(args: Args) -> miette::Result<()> {
 
             if available_tasks.is_empty() {
                 eprintln!("No tasks found",);
+            } else if args.summary {
+                print_heading("Tasks per environment:");
+                for env in project.environments() {
+                    let formatted: String = env.get_filtered_tasks().iter().sorted().fold(
+                        String::new(),
+                        |mut output, name| {
+                            let _ = write!(output, "{}, ", name.fancy_display());
+                            output
+                        },
+                    );
+
+                    println!(
+                        "{:>WIDTH$}: {}",
+                        env.name().fancy_display().bold(),
+                        formatted
+                    );
+                }
+            } else if args.machine_readable {
+                let unformatted: String =
+                    available_tasks
+                        .iter()
+                        .sorted()
+                        .fold(String::new(), |mut output, name| {
+                            let _ = write!(output, "{} ", name.as_str());
+                            output
+                        });
+                println!("{}", unformatted);
             } else {
-                let formatted: String = available_tasks
-                    .iter()
-                    .sorted()
-                    .map(|name| {
-                        if args.summary {
-                            format!("{} ", name.as_str(),)
-                        } else {
-                            format!("* {}\n", name.fancy_display().bold(),)
-                        }
-                    })
-                    .collect();
-
-                println!("{}", formatted);
+                let formatted: String =
+                    available_tasks
+                        .iter()
+                        .sorted()
+                        .fold(String::new(), |mut output, name| {
+                            let _ = write!(output, "{}, ", name.fancy_display());
+                            output
+                        });
+                print_heading("Tasks that can run on this machine:");
+                println!(
+                    "{:>WIDTH$}: {}",
+                    console::Style::new().bold().apply_to("Tasks"),
```

Comment:
```suggestion
                    "{}",
```
I would skip this indent. And the word "tasks"